### PR TITLE
#323: Remove paths duplicates

### PR DESCRIPTION
**Pull request by mvomiero**
Fixes: #323

### CODE REVIEW

File: cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java [hunk 1]
```diff
@@ -177,14 +177,14 @@ public AbstractIdeContext(IdeLogLevel minLogLevel, Function<IdeLogLevel, IdeSubL
...
+
     // detection completed, initializing variables
-    setCwd(userDir, workspace, currentDir);
     Path ideRootPath = null;
     if (currentDir == null) {
       info(getMessageIdeHomeNotFound());
     } else {
       debug(getMessageIdeHomeFound());
-      ideRootPath = this.ideHome.getParent();
+      ideRootPath = currentDir.getParent();
```
Comment (hohwille) [inline cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java:187]: can be `null` here
```suggestion
      if (currentDir != null) {
        ideRootPath = currentDir.getParent();
      }
```

**Review by hohwille — APPROVED** [review]
@mvomiero thanks for your PR. Nice that you have been able to understand the bug and create a solution 👍 
Please avoid NPE then we can merge.

File: cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java (hunk 1 above)
Comment (slskiba) [reply cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java:187]: The check for currentDir being null already happens in line 183 just above

Comment (mvomiero) [reply cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java:187]: https://github.com/devonfw/IDEasy/blob/88385bc05d214fe5c166238ae86eba48bb18be8f/cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java#L183-L187

there is already a check for that a couple of lines before 😉

**Review by slskiba — APPROVED** [review]

File: cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java (hunk 1 above)
Comment (hohwille) [reply cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java:187]: Oh, sorry for the spam. That review was directly before our daily and I completed it in a hurry.
My mistake...

**Pull request merged**